SilKit/ci/check_formatting.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+#
+# SPDX-License-Identifier: MIT
+from pathlib import Path
+from shutil import which
+import re
+import subprocess
+import sys
+
+## Convenience
+def log(fmt, *args):
+    print(fmt.format(*args))
+
+def info(fmt, *args):
+    log("INFO: " + fmt, *args)
+
+def warn(fmt, *args):
+    log("::warning:: " + fmt, *args)
+
+def die(status, fmt, *args):
+    log("ERROR: " + fmt, *args)
+    sys.exit(status)
+
+## Check if clang-format is installed
+
+if which('clang-format') is None:
+    warn("No clang-format found!")
+    die(1, "Please install clang-format!")
+
+# Check for supported clang-format version
+format_version = subprocess.run(['clang-format', '--version'], capture_output=True)
```

Comment:
`subprocess.run` has an optional keyword argument `encoding` which afaik causes the output to be captured as text (i.e., python does the conversions for you). Then the explicit decoding below can be dropped.

Reply:
subprocess.run has way too many args, I just reread the docs and I think I can specify text ouptut with 3 separate flags to do this (encoding, errors, text). So I will roll a dice and see which one of them I will use to make life easier :) thanks

Reply:
Yeah `subprocess` has some historical baggage 😹 🛄